src/ripple/app/misc/SHAMapStoreImp.cpp

```diff
@@ -664,11 +664,11 @@ SHAMapStoreImp::clearPrior (LedgerIndex lastRotated)
                 << deleteBatch << " rows.";
             long long totalRowsAffected = 0;
             long long rowsAffected;
-            soci::statement st = [&]
+            boost::optional<soci::statement> st;
             {
                 auto db = ledgerDb_->checkoutDb();
-                return (db->prepare << deleteQuery);
-            }();
+                st.emplace(db->prepare << deleteQuery);
```

Comment:
The problem with `boost::optional` semantics is that it allows the possibility for a subsequent call to unseat the container. Clearly, behavior we neither want nor need. The lambda is superior, we should use that.